releases/_posts/2025-05-23-spark-release-4-0-0.md

```diff
@@ -0,0 +1,694 @@
+- [[SPARK-50861]](https://issues.apache.org/jira/browse/SPARK-50861) Upgrade dropwizard metrics to 4.2.30
+- [[SPARK-50452]](https://issues.apache.org/jira/browse/SPARK-50452) Upgrade jackson to 2.18.2
+- [[SPARK-48826]](https://issues.apache.org/jira/browse/SPARK-48826) Upgrade fasterxml.jackson to 2.17.2
+- [[SPARK-51006]](https://issues.apache.org/jira/browse/SPARK-51006) Upgrade gcs-connector to 2.2.26
+- [[SPARK-49120]](https://issues.apache.org/jira/browse/SPARK-49120) Bump Gson to 2.11.0
+- [[SPARK-50972]](https://issues.apache.org/jira/browse/SPARK-50972) Upgrade Guava to 33.4.0
+- [[SPARK-49550]](https://issues.apache.org/jira/browse/SPARK-49550) Upgrade Hadoop to 3.4.1
+- [[SPARK-47715]](https://issues.apache.org/jira/browse/SPARK-47715) Upgrade hive‑service‑rpc to 4.0.0
+- [[SPARK-50794]](https://issues.apache.org/jira/browse/SPARK-50794) Upgrade Ivy to 2.5.3
+- [[SPARK-50047]](https://issues.apache.org/jira/browse/SPARK-50047) Upgrade jersey to 3.0.16
+- [[SPARK-49682]](https://issues.apache.org/jira/browse/SPARK-49682) Upgrade joda-time to 2.13.0
+- [[SPARK-47706]](https://issues.apache.org/jira/browse/SPARK-47706) Bump json4s to 4.0.7
+- [[SPARK-50677]](https://issues.apache.org/jira/browse/SPARK-50677) Upgrade jupiter-interface to 0.13.3 and JUnit5 to 5.11.4
+- [[SPARK-50345]](https://issues.apache.org/jira/browse/SPARK-50345) Upgrade Kafka to 3.9.0
+- [[SPARK-50493]](https://issues.apache.org/jira/browse/SPARK-50493) Migrate kubernetes-client from 6.x to 7.x
+- [[SPARK-50580]](https://issues.apache.org/jira/browse/SPARK-50580) Upgrade log4j2 to 2.24.3
+- [[SPARK-49335]](https://issues.apache.org/jira/browse/SPARK-49335) Upgrade Maven to 3.9.9
+- [[SPARK-48625]](https://issues.apache.org/jira/browse/SPARK-48625) Upgrade mssql-jdbc to 12.6.2.jre11
+- [[SPARK-47298]](https://issues.apache.org/jira/browse/SPARK-47298) Upgrade mysql-connector-j to 8.3.0 / mariadb-java-client to 2.7.12
+- [[SPARK-51054]](https://issues.apache.org/jira/browse/SPARK-51054) Upgrade Netty to 4.1.117.Final
+- [[SPARK-50278]](https://issues.apache.org/jira/browse/SPARK-50278) Upgrade netty-tcnative to 2.0.69.Final
+- [[SPARK-45590]](https://issues.apache.org/jira/browse/SPARK-45590) Upgrade okio to 1.17.6
+- [[SPARK-50728]](https://issues.apache.org/jira/browse/SPARK-50728) Update ORC to 2.1.0
+- [[SPARK-50425]](https://issues.apache.org/jira/browse/SPARK-50425) Bump Apache Parquet to 1.15.0
+- [[SPARK-48563]](https://issues.apache.org/jira/browse/SPARK-48563) Upgrade pickle to 1.5
+- [[SPARK-50894]](https://issues.apache.org/jira/browse/SPARK-50894) Postgres driver bump to 42.7.5
+- [[SPARK-50796]](https://issues.apache.org/jira/browse/SPARK-50796) Upgrade protobuf-java to 4.29.3
+- [[SPARK-50821]](https://issues.apache.org/jira/browse/SPARK-50821) Upgrade Py4J from 0.10.9.8 to 0.10.9.9
+- [[SPARK-47737]](https://issues.apache.org/jira/browse/SPARK-47737) Bump PyArrow to 10.0.0
```

Comment:
https://github.com/apache/spark/pull/49282 Upgrade the minimum version of pyarrow to 11.0.0